packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.tsx

```diff
@@ -164,8 +164,11 @@ export class DatePicker extends BaseComponent<IDatePickerProps, IDatePickerState
     let { isDatePickerShown, formattedDate, selectedDate, errorMessage } = this.state;
 
     return (
+      // role='application' is used to work around undesirable behavior in JAWS.
+      // Without role='application', when pressing enter to open the date picker popup,
+      // JAWS swallows the first enter key press and nothing happens.
       <div className={ css('ms-DatePicker', styles.root) } ref='root'>
-        <div ref={ (c): HTMLElement => this._datepicker = c }>
+        <div ref={ (c): HTMLElement => this._datepicker = c } role='application'>
```

Comment:
Thanks for your comments.

I don't think role="dialog" works here because the element the role is being applied to is the container of the text input element, which is not in a dialog.

When you suggested role="input" and role="button", I'm assuming you meant to apply those to the input element (and not the div that contains them). Setting role="input" on an input element should have no effect because it's already an input element.

Setting role="button" on the input element would probably work ok in the case that it's not editable, but in the case that the text input is editable, it would make it appear as the wrong type to the screenreader.